2024-05-10 23:50:59,953 - ERROR - Test 'tests/test_FirstTaskPage.py::test_first_task_first' failed: browser = <selenium.webdriver.chrome.webdriver.WebDriver (session="8c7b69bfbf567e1210d61d4ed2e40a33")>

    def test_first_task_first(browser):
        firsttaskpage = FirstTaskPage(browser)
        # assert firsttaskpage.FindFirstPhrase(), 'Блока "Сила в людях" нету в html коде'
>       assert 0 == 1
E       assert 0 == 1

tests\test_FirstTaskPage.py:7: AssertionError
2024-05-10 23:51:01,961 - INFO - Test 'tests/test_FirstTaskPage.py::test_first_task_second' passed
2024-05-10 23:51:10,767 - ERROR - Test 'tests/test_FirstTaskPageThird.py::test_first_task_third' failed: browser = <selenium.webdriver.chrome.webdriver.WebDriver (session="5d3875a66d5f9d245436ce67f77560b3")>

    def test_first_task_third(browser):
        firsttaskpage = FirstTaskPageThird(browser)
        output = firsttaskpage.SizeOfPictures()
        height = output[0].get_attribute('height')
>       width = output[0].get_attribute('width')

tests\test_FirstTaskPageThird.py:7: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
..\env\Lib\site-packages\selenium\webdriver\remote\webelement.py:178: in get_attribute
    attribute_value = self.parent.execute_script(
..\env\Lib\site-packages\selenium\webdriver\remote\webdriver.py:407: in execute_script
    return self.execute(command, {"script": script, "args": converted_args})["value"]
..\env\Lib\site-packages\selenium\webdriver\remote\webdriver.py:347: in execute
    self.error_handler.check_response(response)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x0000026DF0DE3950>
response = {'status': 404, 'value': '{"value":{"error":"stale element reference","message":"stale element reference: stale elemen...07FF6897A1904]\\n\\tBaseThreadInitThunk [0x00007FFD3DEC257D+29]\\n\\tRtlUserThreadStart [0x00007FFD3E72AA48+40]\\n"}}'}

    def check_response(self, response: Dict[str, Any]) -> None:
        """Checks that a JSON response from the WebDriver does not have an
        error.
    
        :Args:
         - response - The JSON response from the WebDriver server as a dictionary
           object.
    
        :Raises: If the response contains an error message.
        """
        status = response.get("status", None)
        if not status or status == ErrorCode.SUCCESS:
            return
        value = None
        message = response.get("message", "")
        screen: str = response.get("screen", "")
        stacktrace = None
        if isinstance(status, int):
            value_json = response.get("value", None)
            if value_json and isinstance(value_json, str):
                import json
    
                try:
                    value = json.loads(value_json)
                    if len(value) == 1:
                        value = value["value"]
                    status = value.get("error", None)
                    if not status:
                        status = value.get("status", ErrorCode.UNKNOWN_ERROR)
                        message = value.get("value") or value.get("message")
                        if not isinstance(message, str):
                            value = message
                            message = message.get("message")
                    else:
                        message = value.get("message", None)
                except ValueError:
                    pass
    
        exception_class: Type[WebDriverException]
        e = ErrorCode()
        error_codes = [item for item in dir(e) if not item.startswith("__")]
        for error_code in error_codes:
            error_info = getattr(ErrorCode, error_code)
            if isinstance(error_info, list) and status in error_info:
                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)
                break
        else:
            exception_class = WebDriverException
    
        if not value:
            value = response["value"]
        if isinstance(value, str):
            raise exception_class(value)
        if message == "" and "message" in value:
            message = value["message"]
    
        screen = None  # type: ignore[assignment]
        if "screen" in value:
            screen = value["screen"]
    
        stacktrace = None
        st_value = value.get("stackTrace") or value.get("stacktrace")
        if st_value:
            if isinstance(st_value, str):
                stacktrace = st_value.split("\n")
            else:
                stacktrace = []
                try:
                    for frame in st_value:
                        line = frame.get("lineNumber", "")
                        file = frame.get("fileName", "<anonymous>")
                        if line:
                            file = f"{file}:{line}"
                        meth = frame.get("methodName", "<anonymous>")
                        if "className" in frame:
                            meth = f"{frame['className']}.{meth}"
                        msg = "    at %s (%s)"
                        msg = msg % (meth, file)
                        stacktrace.append(msg)
                except TypeError:
                    pass
        if exception_class == UnexpectedAlertPresentException:
            alert_text = None
            if "data" in value:
                alert_text = value["data"].get("text")
            elif "alert" in value:
                alert_text = value["alert"].get("text")
            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here
>       raise exception_class(message, screen, stacktrace)
E       selenium.common.exceptions.StaleElementReferenceException: Message: stale element reference: stale element not found in the current frame
E         (Session info: chrome=124.0.6367.158); For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#stale-element-reference-exception
E       Stacktrace:
E       	GetHandleVerifier [0x00007FF689831562+60802]
E       	(No symbol) [0x00007FF6897AAC62]
E       	(No symbol) [0x00007FF689667CE4]
E       	(No symbol) [0x00007FF68966DE26]
E       	(No symbol) [0x00007FF6896704F6]
E       	(No symbol) [0x00007FF6896FAEC3]
E       	(No symbol) [0x00007FF6896DAB7A]
E       	(No symbol) [0x00007FF6896FA224]
E       	(No symbol) [0x00007FF6896DA923]
E       	(No symbol) [0x00007FF6896A8FEC]
E       	(No symbol) [0x00007FF6896A9C21]
E       	GetHandleVerifier [0x00007FF689B341FD+3217949]
E       	GetHandleVerifier [0x00007FF689B76197+3488183]
E       	GetHandleVerifier [0x00007FF689B6F11F+3459391]
E       	GetHandleVerifier [0x00007FF6898EB926+823622]
E       	(No symbol) [0x00007FF6897B5FFF]
E       	(No symbol) [0x00007FF6897B0F24]
E       	(No symbol) [0x00007FF6897B10B2]
E       	(No symbol) [0x00007FF6897A1904]
E       	BaseThreadInitThunk [0x00007FFD3DEC257D+29]
E       	RtlUserThreadStart [0x00007FFD3E72AA48+40]

..\env\Lib\site-packages\selenium\webdriver\remote\errorhandler.py:229: StaleElementReferenceException
